enclave-runtime/src/lib.rs

```diff
@@ -689,16 +646,17 @@ where
 
 	let mut validator = LightClientSeal::<PB>::unseal()?;
 	let signer = Ed25519Seal::unseal()?;
-	let mut nonce = NONCE.write().expect("Encountered poisoned NONCE lock");
 	let stf_executor = StfExecutor::new(Arc::new(OcallApi), Arc::new(GlobalFileStateHandler));
+	let genesis_hash = validator.genesis_hash(validator.num_relays())?;
+	let extrinsics_factory =
+		ExtrinsicsFactory::new(genesis_hash, signer, GLOBAL_NONCE_CACHE.clone());
```

Comment:
We now get the genesis hash from the light client when we crate an extrinsics factory